packages/api/src/clickhouse/index.ts

```diff
@@ -1148,22 +1148,44 @@ export const buildMetricSeriesQuery = async ({
 
   const rateMetricSource = SqlString.format(
     `
+    SELECT rate, timestamp, _string_attributes, name FROM (
       SELECT
-        if(
-          runningDifference(value) < 0
-          OR neighbor(_string_attributes, -1, _string_attributes) != _string_attributes,
-          nan,
-          runningDifference(value)
-        ) AS rate,
```

Comment:
oh I see. the issue happens when there are counter resets within the window (granularity)